Long-long pull request

Varkh opened the pull request.

kds1660 committed:
change view in progress

1) login/logout work

kds1660 committed:
add topic list

kds1660 committed:
changes for new frontend

remaining page divine and find

kds1660 committed:
fix

kds1660 committed:
add scroll topic

kds1660 committed:
add search

kds1660 committed:
add fixes

kds1660 committed:
scroll and find add

kds1660 committed:
fix

kds1660 committed:
fix scroll at add new topic

kds1660 committed:
edit readme

kds1660 committed:
fix login addcomment

kds1660 committed:
fix at new topic

kds1660 committed:
page=1 in init

kds1660 committed:
add search and sort for find block

kds1660 committed:
refactoring

1) divide topicSchema into 3 schemes and put in folder dbSchema in
modules
2) refactorin engine, putt all request in new module
3) some variables in config folder, divided into prod and dev, prod not
comitted
4) move to test all commands with remove collection and adding it from
data jsons,  now in dev state - restart db with remove collections and
adding from data, in other - no renew db from json files

kds1660 committed:
changes

kds1660 committed:
angular

For now
1) view topics with infinite scroll
2) login/logout

kds1660 committed:
add changes buttons for logged user

kds1660 committed:
remove root controller in separate file

kds1660 committed:
add delete toppic

kds1660 committed:
delete topic+allerts

kds1660 committed:
full login functionality

Varkh commented on modules/dbSchema/topic.js:
```diff
@@ -0,0 +1,34 @@
+var CommentSchema=require('./comment');
+
+var TopicSchema = new mongoose.Schema({
+    name: {
```
@kds1660 How would you handle topics with same name?
Just a question

kds1660 replied:
i cant add topic with existing name, name set in schema as unique

Varkh commented on modules/dbSchema/user.js:
```diff
@@ -0,0 +1,23 @@
+    }
+});
+UserSchema.methods.validPassword = function( pwd ) {
+    return ( this.password === pwd );
```
@kds1660 Read about hashing passwords

kds1660 replied:
this is thing, that i want to do (and passport checking for all requests) but dont do that yet

Varkh commented on public/controllers/rootController.js:
```diff
@@ -0,0 +1,24 @@
+    };
+    $scope.setAllert=function (trueFalse,text) {
+        console.log(trueFalse,text)
+        if (trueFalse===true) {
```
@kds1660 copy-paste?

kds1660 replied:
oops

Varkh commented on public/controllers/loginController.js:
```diff
@@ -0,0 +1,64 @@
+app.controller('loginController',function($scope,$timeout,requestService,buttonService){
```
Create service, move login logic to service. Leave button logic in Controller.

kds1660 replied:
something dont understand - where is login logic and button logic, almost each button = request with action on response, I planned remove all button logic as buttonclick= function () {service.method}

Varkh commented on public/directives/directives.js:
```diff
@@ -0,0 +1,35 @@
+    };
+});
+
+app.directive("loginLogout",function(buttonService) {
```
@kds1660 1 file, 1 directive

Varkh commented on public/index.html:
```diff
@@ -1,45 +1,100 @@
+    <script src="/bower_components/angular/angular.js"></script>
+    <script src="mainApp.js" defer></script>
+
+    <script src="services/services.js" defer></script>
```
@kds1660 I was expecting Gulp usage from you.

Varkh commented on public/scroll.js:
```diff
@@ -0,0 +1,50 @@
+                    img:response1[0][index].img,
+                    comments:response1[0][index].comments
+                };
+                var $copy = Mustache.to_html(template, data);
```
@kds1660 костыли-костыли
time to rewrite jquery and mustache to angular

kds1660 committed:
changes

1. separate directives
2. create Loginservice and Topicservise and put some logic from button
to it
3. View topic and editTopic added
4. remove old files

kds1660 committed:
save before gulp test

kds1660 committed:
add gulp

kds1660 committed:
gulp in devdepencies

kds1660 committed:
fix

kds1660 committed:
add $resources

change all request to $resources usage

kds1660 committed:
+ add topic

kds1660 committed:
comment and search complete

1. add comments  add/del/edit logic
2. add search logic and search parameters

kds1660 committed:
in progress- save point

kds1660 committed:
changes

1. angular-route segmebt add
2. change file structure

kds1660 committed:
changes

1.change aller messages
2. delete unnessesary old parts of code
3. in comment addButton change sve - a to input
4. edit comment view with filter
5. some hrev remove to function with $location
6. when comment view - comment reloaded with request

kds1660 committed:
passport check requests and encript password in db
